notebooks/config.ipynb

```diff
@@ -0,0 +1,1046 @@
+    "<!--cell 22-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "4f237a73-da83-4632-990f-644632b15cd9",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "greeting: Hello\n",
+      "recipient: World\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat tmp/config1.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "902c4863-e772-49c6-88f2-31e1ab76b418",
+   "metadata": {},
+   "source": [
+    "Input and output formats are not required to match. This can be used to convert a config from one format to another.\n",
+    "<!--cell 24-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "b8854dc6-9dd2-4843-99e4-278b116b9767",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'greeting': 'Hello', 'recipient': 'World'}"
+      ]
+     },
+     "execution_count": 13,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "config.realize(\n",
+    "    input_config='fixtures/config/get-config.yaml',\n",
+    "    input_format='yaml',\n",
+    "    output_file='tmp/realize-config.sh',\n",
+    "    output_format='sh'\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4032ab38-f6c7-45d1-bff7-f70d23832f26",
+   "metadata": {},
+   "source": [
+    "Here a Shell config is created from a YAML config.\n",
+    "<!--cell 26-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "0a59a3e8-27b5-4daa-a924-941aceaad157",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "greeting=Hello\n",
+      "recipient=World\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat tmp/realize-config.sh"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ef0136e4-f549-467f-9341-f77f84738bb0",
+   "metadata": {},
+   "source": [
+    "### Updating configs\n",
+    "\n",
+    "Configs can be updated by providing a second config with the `update_config` parameter. If the update config contains keys that match the base config, they will be overwritten. If the base config contains Jinja2 expressions, like the one below, they will be rendered using the values in the update config wherever possible.\n",
+    "<!--cell 28-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "2d72bbc1-e438-48b2-8bd7-554b598c6f24",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "&memo\n",
+      "  sender_id = \"{{ id }}\"\n",
+      "  message = \"{{ greeting }}, {{ recipient }}!\"\n",
+      "  sent = .FALSE.\n",
+      "/\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat fixtures/config/update-config.nml"
```

Comment:
Does it make sense to name this one `base-config.nml`?

Reply:
I agree, this name and the one below were confusing. I updated them both.